tests/unit/s2n_client_status_request_extension_test.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/extensions/s2n_client_status_request.h"
+#include "tls/s2n_resume.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    struct s2n_config *config;
+    EXPECT_NOT_NULL(config = s2n_config_new());
+
+    /* Test should_send */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+        /* status request should be sent by default */
+        EXPECT_FALSE(s2n_client_status_request_extension.should_send(conn));
+
+        /* status request should be sent if ocsp requested */
+        conn->config->status_request_type = S2N_STATUS_REQUEST_OCSP;
+        EXPECT_TRUE(s2n_client_status_request_extension.should_send(conn));
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test send */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+        struct s2n_stuffer stuffer;
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+
```

Comment:
The default value for config->status_request_type is set to `S2N_STATUS_REQUEST_NONE` [s2n_config.c#L91](https://github.com/awslabs/s2n/blob/master/tls/s2n_config.c#L91), shouldnt we be setting config->status_request_type to `S2N_STATUS_REQUEST_OCSP` prior to send?

Reply:
We were sneakily inheriting it from the should_send test setup, but I can re-set the value more explicitly.